packages/amplify-codegen-model-plugin/src/languages/swift-declaration-block.ts

```diff
@@ -0,0 +1,302 @@
+import { indent, indentMultiline } from '@graphql-codegen/visitor-plugin-common';
+import { NameNode, StringValueNode } from 'graphql';
+
+function isStringValueNode(node: any): node is StringValueNode {
+  return node && typeof node === 'object' && node.kind === 'StringValue';
+}
+
+function transformComment(comment: string | StringValueNode, indentLevel = 0): string {
+  if (!comment || comment === '') {
+    return '';
+  }
+
+  if (isStringValueNode(comment)) {
+    comment = comment.value;
+  }
+
+  const lines = comment.split('\n');
+
+  return lines
+    .map((line, index) => {
+      const isLast = lines.length === index + 1;
+      const isFirst = index === 0;
+
+      if (isFirst && isLast) {
+        return indent(`// ${comment} \n`, indentLevel);
+      }
+      line = line.split('*/').join('*\\/');
+      return indent(`${isFirst ? '/*' : ''} * ${line}${isLast ? '\n */\n' : ''}`, indentLevel);
+    })
+    .join('\n');
+}
+
+export type Access = 'private' | 'public' | 'DEFAULT';
+export type VariableFlags = {
+  isList?: boolean;
+  variable?: boolean;
+};
+export type StructFlags = VariableFlags & { optional?: boolean; static?: boolean };
+export type PropertyFlags = StructFlags;
+export type MethodFlags = { static?: boolean };
+export type DeclarationFlag = { final?: boolean };
+export type Kind = 'class' | 'struct' | 'extension' | 'enum';
+export type VariableDeclaration = {
+  value: string | undefined;
+  name: string;
+  type: string;
+  flags: VariableFlags;
+};
+export type StructProperty = VariableDeclaration & {
+  access: Access;
+  comment?: string;
+  getter?: string;
+  setter?: string;
+  flags: PropertyFlags;
+};
+export type MethodArgument = VariableDeclaration & {
+  flags: PropertyFlags;
+};
+export type StructMethod = {
+  args: MethodArgument[];
+  implementation: string;
+  name: string;
+  access: Access;
+  returnType: string | null;
+  flags: MethodFlags;
+  comment: string;
+};
+export class SwiftDeclarationBlock {
+  _name: string = '';
+  _kind: Kind = 'struct';
+  _protocols: string[] = [];
+  _access: Access = 'DEFAULT';
+  _properties: StructProperty[] = [];
+  _methods: StructMethod[] = [];
+  _comment: string = '';
+  _block: string[] = [];
+  _enumValues: { [name: string]: string } = {};
+  _flags: DeclarationFlag = { final: false };
+
+  access(access: Access): SwiftDeclarationBlock {
+    this._access = access;
+
+    return this;
+  }
+
+  final(isFinal: boolean = true): SwiftDeclarationBlock {
+    this._flags.final = isFinal;
+    return this;
+  }
+
+  withComment(comment: string | StringValueNode | null): SwiftDeclarationBlock {
```

Comment:
Generic question: shouldn't we guard against immutability while building the object? What I mean is check if ```withComment``` or ```withName``` was previously invoked.

Reply:
I don't think its worth it much check this.